internal/controllers/topology/cluster/desired_state.go

```diff
@@ -611,6 +626,7 @@ func computeMachineDeploymentVersion(s *scope.Scope, desiredControlPlaneState *s
 	// Check if we are about to upgrade the control plane. In that case, do not upgrade the machine deployment yet.
 	// Wait for the new upgrade operation on the control plane to finish before picking up the new version for the
 	// machine deployment.
+	// TODO: We probably don't need this check anymore.
```

Comment:
I think you might be right. But because it's very easy to miss an edge case I would prefer keeping this logic as a fail safe so that we never under any circumstances trigger a MD rollout while the CP is still upgrading.

I think with the current code it's also easier to assess that vs. if we have to infer that based on the code above for all edge cases that we can think of.